### ISSUE TITLE
Unreachable code after exception is thrown in kotlin elvis operator

### ISSUE THREAD

**Issue by DcortezMeleth** [detekt/detekt#3495]
## Expected Behavior
When seeing code like I wouldn't expect to see Detekt reporting UnreachableCode for this method.  
```
    private fun getAuthenticatedUserDetails(userId: String): Maybe<User> =
        userService.getUserName(userId)
                .map { userData ->
                    val userName = userData.userName
                            ?: throw UnauthenticatedException("Unable to find user")

                    User(userId, userName)
                }
```

## Observed Behavior
Detekt is reporting UnreachableCode for this method.  

## Steps to Reproduce
Reported code:
```
    private fun getAuthenticatedUserDetails(userId: String): Maybe<User> =
        userService.getUserName(userId)
                .map { userData ->
                    val userName = userData.userName
                            ?: throw UnauthenticatedException("Unable to find user")

                    User(userId, userName)
                }
```
Detekt config for UnreachableCode:
```
potential-bugs:
  active: true
  UnreachableCode:
    active: true
```

## Context
When upgrading Detekt gradle plugin from version 1.0.1 to 1.15.0 build started to fail because of many false-positives returned by UnreachableCode rule.

## Your Environment

* Version of detekt used: 1.15.0
* Version of Gradle used (if applicable): 6.4.
* Gradle scan link (add `--scan` option when running the gradle task):
* Operating System and version: MacOS Big Sur 11.1 (20C69)
* Link to your project (if it's a public repository):

**Comment by cortinico** [comment]
Could be related to https://github.com/detekt/detekt/pull/3220

**Comment by marschwar** [comment]
Is there any way you can share some more of the surrounding code? Unfortunately I have not been able to reproduce the issue.

**Comment by DcortezMeleth** [comment]
I will try to provide failing example. Unfortunately the code is not my property so I cannot link whole project here.
To be frank I also have tried to reproduce it in UnreachableCode Spec but wasn't able to do so.

**Comment by schalkms** [comment]
I couldn't reproduce the issue either.
@DcortezMeleth do you have type solving enabled, since I suspect that this causes the issue? 
At the moment, there's nothing we can really do here.

**Comment by BraisGabin** [comment]
Closing for now. If this issue persists after #3220 please re-open the issue.
